Tests/Tests/CDTSessionCookieInterceptorTests.m

```diff
@@ -0,0 +1,105 @@
+//
+//  CDTCookieInterceptorTests.m
+//  Tests
+//
+//  Created by Rhys Short on 09/09/2015.
+//  Copyright (c) 2015 IBM Corp.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+//
+
+#import <XCTest/XCTest.h>
+#import "CloudantSyncTests.h"
+#import "CDTSessionCookieInterceptor.h"
+#import <OHHTTPStubs/OHHTTPStubs.h>
+// expose properties so we can look at them
+@interface CDTSessionCookieInterceptor ()
+
+@property (nonatomic) BOOL shouldMakeCookieRequest;
+@property (nullable, strong, nonatomic) NSString *cookie;
+@property (nonnull, nonatomic, strong) NSURLSession *urlSession;
+
+@end
+
+@interface CDTSessionCookieInterceptorTests : CloudantSyncTests
+
+@end
+
+@implementation CDTSessionCookieInterceptorTests
+
+- (void)setUp
+{
+    [OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest *__nonnull request) {
+      return [[request.URL host] isEqualToString:@"username1.cloudant.com"];
+    }
+        withStubResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+          return [OHHTTPStubsResponse responseWithJSONObject:@{} statusCode:401 headers:@{}];
+        }];
+
+    [OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest *__nonnull request) {
+      return [[request.URL host] isEqualToString:@"username.cloudant.com"];
+    }
+        withStubResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+          return [OHHTTPStubsResponse
+              responseWithJSONObject:@{
+                  @"ok" : @(YES),
+                  @"name" : @"username",
+                  @"roles" : @[ @"_admin" ]
+              }
+                          statusCode:200
+                             headers:@{
+                                 @"Set-Cookie" :
+                                     @"AuthSession=cm9vdDo1MEJCRkYwMjq0LO0ylOIwShrgt8y-UkhI-c6BGw; "
+                                     @"Version=1; Path=/; HttpOnly"
+                             }];
+        }];
+}
+
+- (void)tearDown { [OHHTTPStubs removeAllStubs]; }
+- (void)testCookieInterceptorSuccessfullyGetsCookie
+{
+    NSString *expectedCookieString = @"AuthSession=cm9vdDo1MEJCRkYwMjq0LO0ylOIwShrgt8y-UkhI-c6BGw";
```

Comment:
You want to make this a constant so you can share it between the response and the expection?